cpp/src/skyhook/protocol/rados_protocol.h

```diff
@@ -0,0 +1,165 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+#pragma once
+
+#include <rados/librados.hpp>
+
+#include "arrow/status.h"
+
+#include "skyhook/client/file_skyhook.h"
+
+namespace skyhook {
+namespace rados {
+
+/// Wrap Arrow Status with a custom return code.
+class RadosStatus {
+ public:
+  RadosStatus(arrow::Status s, int code) : s_(s), code_(code) {}
+  arrow::Status status() { return s_; }
+  int code() { return code_; }
+
+ private:
+  arrow::Status s_;
+  int code_;
+};
+
+class IoCtxInterface {
```

Comment:
Is there a need to have interfaces when there seems to be only one possible implementation? pImpl might be better suited

Reply:
Since, rados protocol and skyhook protocol is not meant to be exposed publicly, maybe doing PIMPL would be a overkill.